google/cloud/forseti/scanner/scanners/firewall_rules_scanner.py

```diff
@@ -176,10 +181,22 @@ def _retrieve(self):
             list: List of firewall policy data.
             int: The resource count.
         """
-        firewall_policies = (firewall_rule_dao
-                             .FirewallRuleDao(self.global_configs)
-                             .get_firewall_rules(self.snapshot_timestamp))
 
+        model_manager = self.service_config[0].model_manager
+        scoped_session, data_access = model_manager.get(self.model_name)
+        with scoped_session as session:
+            firewall_policies = []
+
+            for i in data_access.scanner_iter(session, "firewall"):
+                firewall_data_for_scanner = ast.literal_eval(i.data)
```

Comment:
That won't work, it will die for data other than the firewall (which happens to not have any Boolean values so doesn't crash). It is JSON data, it just isn't a json type. You have to load it with json.loads.